File: ZooJava/src/main/java/com/example/zoojava/repositories/animals/AnimalsRepository.java
package com.example.zoojava.repositories.animals;

import com.example.zoojava.models.Animal;
import com.example.zoojava.repositories.ICRUD;
import javafx.collections.ObservableList;

import java.sql.SQLException;
import java.util.List;

public interface AnimalsRepository extends ICRUD<Animal> {
    ObservableList<Animal> findAll();
    Animal findById(int id);
    Animal modifyById(int id, Animal newAnimal) throws SQLException;
}

File: ZooKotlin/src/main/kotlin/com/example/zookotlin/repositories/animals/AnimalsRepository.kt
package com.example.zookotlin.repositories.animals

import com.example.zookotlin.models.Animal
import com.example.zookotlin.repositories.ICRUD
import javafx.collections.ObservableList

interface AnimalsRepository: ICRUD<Animal> {
    fun findAll(): ObservableList<Animal>
    fun findById(id: Int): Animal
    fun modifyById(id: Int, newAnimal: Animal):Animal?
}